pkg/auth/providers/common/usermanager.go

```diff
@@ -265,16 +266,23 @@ func (m *userManager) EnsureUser(principalName, displayName string) (*v3.User, e
 		// User doesn't exist, create user
 		logrus.Infof("Creating user for principal %v", principalName)
 
+		// Create a hash of the principalName to use as the name for the user,
+		// this lets k8s tell us if there are duplicate users with the same name
+		// thus avoiding a race.
+		hasher := sha256.New()
+		hasher.Write([]byte(principalName))
+		sha := base32.StdEncoding.WithPadding(-1).EncodeToString(hasher.Sum(nil))[:10]
```

Comment:
will this panic if length of the encoding string is less than 10? if so, can we protect against that?

Reply:
Technically, yes, but sha256 always returns the same length of byte array so we are always encoding to the same length of string (52) regardless of the length of the input string.